File: mobile_app/lib/models/question.dart
class Question {
  final int id;
  final String questionText;
  final String category;
  final String type;
  final bool isActive;

  Question({
    required this.id,
    required this.questionText,
    required this.category,
    required this.type,
    required this.isActive,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'],
      questionText: json['question_text'],
      category: json['category'],
      type: json['type'],
      isActive: json['is_active'] == 1,
    );
  }
}

File: backend/app/Models/Question.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Question extends Model
{
    use HasFactory;

    protected $fillable = [
        'question_text',
        'category',
        'type',
        'is_active',
    ];

    public function responseItems()
    {
        return $this->hasMany(ResponseItem::class);
    }

    public function assessments()
    {
        return $this->belongsToMany(Assessment::class);
    }
}
